src/features.ts

```diff
@@ -0,0 +1,56 @@
+import * as core from '@actions/core'
+import * as tc from '@actions/tool-cache'
+import {IS_LINUX} from './constants'
+import {exec} from '@actions/exec'
+import {homedir} from 'os'
+import {join} from 'path'
+import {mkdirP} from '@actions/io'
+
+export async function enableFeatures(features: string[]): Promise<void> {
+  for (const feature of features) {
+    await enableFeature(feature)
+  }
+}
+
+async function enableFeature(feature: string): Promise<void> {
+  switch (feature) {
+    case 'native-image-musl':
+      await enableNativeImageMusl()
+      break
+    default:
+      core.warning(`Unknown feature requested: ${feature}`)
+      break
+  }
+}
+
+async function enableNativeImageMusl(): Promise<void> {
+  if (!IS_LINUX) {
+    core.warning('musl is only supported on Linux')
+    return
+  }
+  core.startGroup(`Setting up musl for GraalVM Native Image...`)
+  const basePath = join(homedir(), '.musl_feature')
+  await mkdirP(basePath)
+
+  const muslName = 'x86_64-linux-musl-native'
+  const muslDownloadPath = await tc.downloadTool(
+    `http://more.musl.cc/10/x86_64-linux-musl/${muslName}.tgz`
+  )
+  await tc.extractTar(muslDownloadPath, basePath)
+  const muslPath = join(basePath, muslName)
+  core.addPath(join(muslPath, 'bin'))
+  core.exportVariable('CC', join(muslPath, 'bin', 'gcc'))
```

Comment:
I'm not very familiar with the github actions API, but could we unset CC after we compile and install zlib? As the variable is only needed to build zlib, we should be able to remove it afterwards

Reply:
Thanks for pointing that out! I've dropped the export of `CC` and added it to the zlib build commands instead.